torch/csrc/distributed/rpc/request_callback_impl.cpp

```diff
@@ -137,15 +137,29 @@ std::unique_ptr<RpcCommandBase> RequestCallbackImpl::processRpc(
       // Attach 'recv' autograd function.
       DistAutogradContext* autogradContext = addRecvRpcBackward(
           rpcWithAutograd.autogradMetadata(), rpcWithAutograd.tensors());
+      // For this recv thread, before processRpc(), set current_context_id_ to
+      // be context_id passed from client. In this way, if there is nested
+      // rpc call in python rpc call, original context_id from client can be
+      // passed in the chain calls.
+      auto& autogradContainer = DistAutogradContainer::getInstance();
+      if (autogradContext != nullptr) {
```

Comment:
is it expected to see `MESSAGE_WITH_AUTOGRAD_REQ` if there is no valid context?

Reply:
autogradContext == nullptr here means: no valid context or tensors do not require grads